### PR TITLE
Change all .format() instances to f-strings

### PR DESCRIPTION
**Pull request by mariaschuld**
**Context:**

Presumably due to changes in the PEP standard, covecov is starting to complain about our use of ``"my string {}".format(i)`` instead of ``f"my string {i}"``. 

**Description of the Change:**

This PR changes all code instances of format into f-strings.

**Benefits:**

Hopefully no one else has to worry

### CODE REVIEW

File: pennylane/_grad.py [hunk 1]
```diff
@@ -445,6 +439,6 @@ def finite_diff(f, N=1, argnum=0, idx=None, delta=0.01):
         return partial(_fd_second_order_centered, f, argnum, delta, idx=idx)
 
     raise ValueError(
-        "At present, finite-difference approximations are supported up to second-order."
-        " The value of 'N' can be 1 or 2; got {}".format(N)
+        f"At present, finite-difference approximations are supported up to second-order."
+        " The value of 'N' can be 1 or 2; got {N}"
```
Comment (mariaschuld) [inline pennylane/_grad.py:443]:
```suggestion
        f" The value of 'N' can be 1 or 2; got {N}"
```

File: pennylane/measure.py [hunk 2]
```diff
@@ -91,13 +91,13 @@ def diagonalizing_gates(self):
...
-            return "{}(wires={})".format(self.return_type.value, self.wires.tolist())
+            return f"{self.return_type.value}(wires={self.wires.tolist()})"
 
         # Todo: when tape is core the return type will always be taken from the MeasurementProcess
         if self.obs.return_type is None:
-            return "{}({})".format(self.return_type.value, self.obs)
+            return f"{self.return_type.value}({self.obs})"
 
-        return "{}".format(self.obs)
+        return "{self.obs}"
```
Comment (mariaschuld) [inline pennylane/measure.py:100]:
```suggestion
        return f"{self.obs}"
```

File: tests/test_device.py [hunk 3]
```diff
@@ -339,7 +339,7 @@ def test_check_validity_with_not_supported_operation_inverse(
 
         with pytest.raises(
             DeviceError,
-            match="Gate {} not supported on device {}".format("CNOT", "MockDevice"),
```
Comment (KetpuntoG) [inline tests/test_device.py:342]: I wonder why it got that way back in the day? 🤔

**Review by KetpuntoG — APPROVED** [review]
I always liked the f-string format better 😄

File: tests/test_device.py (hunk 3 above)
Comment (mariaschuld) [reply tests/test_device.py:342]: Copy and paste from the main code, and replace the variables with fixed values :)

**Comment by mariaschuld** [comment]
Codecov complains because I split one untested line (the output of `about`) into two. Will overwrite.

**Pull request merged**